src/utils/client-factory.ts

```diff
@@ -52,8 +55,36 @@ const clientFactory = <T extends ClientType>(
 
         case 'ApolloGraphQL':
             return new ApolloClient({
+                // https://www.apollographql.com/docs/react/performance/server-side-rendering#initializing-apollo-client
+                ssrMode: true, // prevents ApolloClient from refetching queries unnecessarily
                 cache: new InMemoryCache(),
                 uri,
+                headers,
+                link:
+                    // https://www.apollographql.com/docs/react/networking/advanced-http-networking/#overriding-options
+                    new HttpLink({
+                        uri,
+                        headers,
+                        fetchOptions: { method: 'GET' }, // override default POST to use GET
+                        // log uri (on dev mode) and fetch
+                        fetch: (...pl) => {
+                            if (process.env.NODE_ENV === 'production') {
+                                return fetch(...pl);
+                            }
+
+                            // https://github.com/apollographql/apollo-client/issues/4017
+                            // tweaked from musemind implementation
+                            const [uri] = pl;
+
+                            // because queries are long,
+                            // so save them locally instead of logging to console
+                            appendFileSync(
+                                'gql_uri_log.txt',
+                                decodeURI(uri as string)
+                            );
+                            return fetch(...pl);
+                        },
```

Comment:
@harikakotipalli Want to alert you on this and let me know if this is unnecessary